### ISSUE TITLE
Performance improvement need for GRPC storage plugin

### ISSUE THREAD

**Issue by Sreevani871** [jaegertracing/jaeger#2357]
In our project we have written grpc storage plugin. When we perform load test, grpc plugin performance is very poor compare to ES Storage type. And we even tested with binary built by adding plugin code as a new storage type in jaeger codebase , its performance is par with ES storage type.

Attaching here screenshots of load test metrics with collector+grpc plugin and collector binary built by adding grpc plugin code as a new datastore in jaeger codebase.

**collector+grpc**
80% spans received are dropped

<img width="1353" alt="grpc1" src="https://user-images.githubusercontent.com/20114703/88269909-7973b700-ccf2-11ea-828c-99c059ddf3ea.png">
<img width="1367" alt="grpc2" src="https://user-images.githubusercontent.com/20114703/88269929-7e386b00-ccf2-11ea-8b8c-e234f9824c5c.png">
<img width="697" alt="Screen Shot 2020-07-23 at 2 28 00 PM" src="https://user-images.githubusercontent.com/20114703/88269932-80022e80-ccf2-11ea-8a39-71e06fe01fbd.png">

**collector binary built by adding plugin code as a new datastore in jaeger codebase.**
no spans drops
<img width="1370" alt="single binary" src="https://user-images.githubusercontent.com/20114703/88270169-dbccb780-ccf2-11ea-82b9-6cecbb5673c8.png">
<img width="1380" alt="single binary 2" src="https://user-images.githubusercontent.com/20114703/88270189-e0916b80-ccf2-11ea-8292-6c2507371b55.png">

Is there any way we can improve grpc plugin performance?

**Comment by yurishkuro** [comment]
Even though the grpc plugin architecture introduces an extra RPC roundtrip and proto serialization, I am quite skeptical that it would result in 100ms latency since the communication happens over local unix pipe. The absolutely flat 100ms latency line is also highly suspicious, looks like saturation of throttling of some kind.

I don't know why this is, but I don't think this behavior is purely to how the plugin is architected, there seems to be something else going on here. Try to investigate further. Add additional metrics to the plugin itself, e.g. what is the latency of actual storage operations, how many concurrent saves are being processed, etc.

**Comment by Sreevani871** [comment]
Yes 100 ms is what we worry about, We even tested with returning nil from WriteSpan() method at plugin side without any operations going on still we see the same performance with grpc plugin. 
Hardaware Details:
4 CPU's
8 GB RAM
